crates/edr_generic/src/rpc/transaction.rs

```diff
@@ -0,0 +1,113 @@
+use edr_eth::{transaction::SignedTransaction as _, SpecId};
+use edr_evm::{
+    block::transaction::{BlockDataForTransaction, TransactionAndBlock},
+    transaction::remote::EthRpcTransaction,
+};
+use edr_rpc_eth::RpcTypeFrom;
+use serde::{Deserialize, Serialize};
+
+use crate::GenericChainSpec;
+
+// We need to use a newtype here as `RpcTypeFrom` cannot be implemented here,
```

Comment:
In general there were few cases where it was not possible to re-use the original traits as we did depend on the actual trait implementation or the associated type was set for a given impl, rather than be generic (and so pluggable by potential other impls, like `(Fake)Sign`).

How often that's the case I'm not sure, as this was a very specific case of "do everything that L1 does except these few tiny bits" but I'd assume that we should provide sensible defaults or building blocks that the other implementers can more easily re-use what we have rather than copy the impls in some cases.

What's the ergonomic solution I'm not yet sure and I'm mostly just thinking out loud. Maybe allow for more generic impls and use the final marker trait impls that tie all of these together to validate that the intermediate types are compatible and expected?

For instance, I found it hard to wrap my head around what to implement as often the traits felt like they were circular, like in the case of satisfying both the `RpcSpec` and `ChainSpec` for a type. Another case is that some types like `EthBlockData` only use a single associated type from the `ChainSpec` and the `ChainSpec` feels like this tie-all/validation trait (which makes sense) but it made me harder to understand what's actually used and why. I had trouble deciphering some of the compiler errors and figuring out which trait impls are required because it felt so intertwined.

Maybe I just missed a proper mental model but I have to admit that it took me longer than it should :see_no_evil: